src/gravel/controllers/orch/orchestrator.py

```diff
@@ -36,35 +38,71 @@ class Orchestrator:
     def __init__(self, ceph_mgr: Mgr):
         self.cluster = ceph_mgr
 
-    def call(self, cmd: Dict[str, Any]) -> Any:
+    def call(self, cmd: Dict[str, Any], inbuf: bytes = b"") -> Any:
         if "format" not in cmd:
             cmd["format"] = "json"
-        return self.cluster.call(cmd)
+        return self.cluster.call(cmd, inbuf=inbuf)
 
     def host_ls(self) -> List[OrchHostListModel]:
         cmd = {"prefix": "orch host ls"}
         res = self.call(cmd)
         return parse_obj_as(List[OrchHostListModel], res)
 
-    def devices_ls(self) -> List[OrchDevicesPerHostModel]:
-        cmd = {"prefix": "orch device ls"}
+    def devices_ls(
+        self,
+        hostname: Optional[str] = None
+    ) -> List[OrchDevicesPerHostModel]:
+        cmd: Dict[str, Any] = {"prefix": "orch device ls"}
+        if hostname and len(hostname) > 0:
+            cmd["hostname"] = [hostname]
+
         res = self.call(cmd)
         return parse_obj_as(List[OrchDevicesPerHostModel], res)
 
-    def assimilate_all_devices(self) -> None:
+    def assimilate_devices(self, host: str, devices: List[str]) -> None:
+        spec = {
+            "service_type": "osd",
+            "service_id": "default_drive_group",
+            "placement": {
+                "hosts": [host]
+            },
+            "data_devices": {
+                "paths": devices
+            }
+        }
+        specstr: Optional[str] = None
+        try:
+            specstr = yaml.dump(spec)  # type: ignore
+        except Exception as e:
+            raise GravelError(str(e))
+        assert specstr is not None
```

Comment:
```suggestion
        specstr: str = yaml.dump(spec)
```

Reply:
I don't think this needs exception handling? It also appears as if `yaml.dump` will always produce a str type..

Reply:
good call. It may return `None` however, so I'll keep the assert, even if it's just for static type checker purposes.